pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -142,7 +142,12 @@ func makeVirtualServiceSpec(ia v1alpha1.IngressAccessor, gateways map[v1alpha1.I
 
 // qualifyGateways modifies the provided gateway list to add namespace
 // information into gateway names that don't already have them.
-func qualifyGateways(gws []string) []string {
+func qualifyGateways(gwsOrg []string) []string {
+	if len(gwsOrg) == 0 {
+		return nil
+	}
+	gws := make([]string, len(gwsOrg))
+	copy(gws, gwsOrg)
```

Comment:
If we have to copy this, then I guess just copy in the loop. No need to copy the object and then replace it.